src/validation.cpp

```diff
@@ -106,6 +108,20 @@ const std::vector<std::string> CHECKLEVEL_DOC {
     "level 4 tries to reconnect the blocks",
     "each level includes the checks of the previous levels",
 };
+inline std::string_view assumevalid_reason(const AssumeValid v) noexcept
+{
+    using enum AssumeValid;
+    switch (v) {
+    case SKIPPED:                       return "";
+    case CHECKED:                       return "assumevalid=0 (always verify)";
+    case CHECKED_HASH_NOT_IN_HEADERS:   return "assumevalid hash not in headers";
+    case CHECKED_NOT_UNDER_ASSUMEVALID: return "block not under assumevalid anchor";
+    case CHECKED_OFF_BESTHEADER_PATH:   return "block not on best header path";
+    case CHECKED_BELOW_MIN_CHAINWORK:   return "best header chainwork below nMinimumChainWork";
```

Comment:
nit: logs shouldn’t expose internal variable names

```suggestion
    case CHECKED_BELOW_MIN_CHAINWORK:   return "best header chainwork below -minimumchainwork";
```